ext/opentelemetry-ext-testutil/src/opentelemetry/ext/testutil/wsgitestutil.py

```diff
@@ -1,32 +1,28 @@
 import io
 import unittest
-import unittest.mock as mock
 import wsgiref.util as wsgiref_util
 
 from opentelemetry import trace as trace_api
+from opentelemetry.sdk.trace import TracerSource, export
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+    InMemorySpanExporter,
+)
 
 
 class WsgiTestBase(unittest.TestCase):
-    def setUp(self):
-        self.span = mock.create_autospec(trace_api.Span, spec_set=True)
-        tracer = trace_api.Tracer()
-        self.get_tracer_patcher = mock.patch.object(
-            trace_api.TracerSource,
-            "get_tracer",
-            autospec=True,
-            spec_set=True,
-            return_value=tracer,
-        )
-        self.get_tracer_patcher.start()
-
-        self.start_span_patcher = mock.patch.object(
-            tracer,
-            "start_span",
-            autospec=True,
-            spec_set=True,
-            return_value=self.span,
+    @classmethod
+    def setUpClass(cls):
+        trace_api.set_preferred_tracer_source_implementation(
+            lambda T: TracerSource()
         )
-        self.start_span = self.start_span_patcher.start()
+
+    def setUp(self):
+        tracer_source = trace_api.tracer_source()
+
+        self.memory_exporter = InMemorySpanExporter()
+        span_processor = export.SimpleExportSpanProcessor(self.memory_exporter)
+        tracer_source.add_span_processor(span_processor)
```

Comment:
tracer_source would be shared by all test cases so you will be adding new span processor for every test case executed, maybe using a single InMemorySpanExporter and just call clear method during setup?

Reply:
Updated the test to set a single in-memory span exporter on class setup and clear it each time.